cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/KafkaConnectService.java

```diff
@@ -0,0 +1,176 @@
+    ResponseEntity<String> responseNew;
+    try {
+      responseNew =
+          reqDetails.getRight().postForEntity(reqDetails.getLeft(), request, String.class);
+    } catch (RestClientException e) {
+      log.error("Error in registering new connector " + e.toString());
+      throw new Exception(e.toString());
+    }
+    if (responseNew.getStatusCodeValue() == 201) {
+      return ApiResponse.builder().result(ApiResultStatus.SUCCESS.value).build();
+    } else {
+      return ApiResponse.builder().result(ApiResultStatus.FAILURE.value).build();
+    }
+  }
+
+  public List<String> getConnectors(String environmentVal, KafkaSupportedProtocol protocol) {
+    try {
+      log.info("Into getConnectors {} {}", environmentVal, protocol);
+      if (environmentVal == null) {
+        return null;
+      }
+
+      String suffixUrl = environmentVal + "/connectors";
+      Pair<String, RestTemplate> reqDetails =
+          clusterApiUtils.getRequestDetails(suffixUrl, protocol, KafkaClustersType.KAFKA_CONNECT);
+
+      Map<String, String> params = new HashMap<>();
+      ResponseEntity<List> responseList =
+          reqDetails.getRight().getForEntity(reqDetails.getLeft(), List.class, params);
+
+      log.info("connectors list " + responseList);
+      return responseList.getBody();
+    } catch (Exception e) {
+      log.error("Error in getting connectors " + e.getMessage());
+      return new ArrayList<>();
+    }
+  }
+
+  public LinkedHashMap<String, Object> getConnectorDetails(
+      String connector, String environmentVal, KafkaSupportedProtocol protocol) {
+    try {
+      log.info("Into getConnectorDetails {} {}", environmentVal, protocol);
+      if (environmentVal == null) {
+        return null;
+      }
+
+      String suffixUrl = environmentVal + "/connectors" + "/" + connector;
+      Pair<String, RestTemplate> reqDetails =
+          clusterApiUtils.getRequestDetails(suffixUrl, protocol, KafkaClustersType.KAFKA_CONNECT);
+
+      Map<String, String> params = new HashMap<>();
+
+      ResponseEntity<LinkedHashMap> responseList =
+          reqDetails.getRight().getForEntity(reqDetails.getLeft(), LinkedHashMap.class, params);
+      log.info("connectors list " + responseList);
+
+      return responseList.getBody();
+    } catch (Exception e) {
+      log.error("Error in getting connector detail " + e.getMessage());
+      return new LinkedHashMap<>();
+    }
+  }
+
+  protected ClusterStatus getKafkaConnectStatus(
+      String environment, KafkaSupportedProtocol protocol) {
+    String suffixUrl = environment + "/connectors";
+    Pair<String, RestTemplate> reqDetails =
+        clusterApiUtils.getRequestDetails(suffixUrl, protocol, KafkaClustersType.KAFKA_CONNECT);
+    Map<String, String> params = new HashMap<String, String>();
+
+    try {
+      reqDetails.getRight().getForEntity(reqDetails.getLeft(), Object.class, params);
+      return ClusterStatus.ONLINE;
+    } catch (RestClientException e) {
+      e.printStackTrace();
```

Comment:
```suggestion
      log.error("Exception:", e);
```
it's better to log in logger rather than system output